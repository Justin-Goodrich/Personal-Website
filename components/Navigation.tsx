import React, {FC} from "react"

const nav = [
    {
        name:"home",
        icon:"",
        url:"/"
    },
    {
        name: "skills",
        icon:"",
        url:""
    },
    {
        name: "projects",
        icon:"",
        url:""
    },
    {
        name: "contact",
        icon:"",
        url:""
    }
]
const Navigation:FC  = () => {
    return (
      <nav className="hidden md:flex w-fit p-0.5 border-black border-2 rounded-3xl space-x-3 items-center 2xl:border-4 2xl:p-2 2xl:rounded-full">
        {
            nav.map((link,index)=>{
                return <button className="hidden border-black border-2 rounded-3xl text-2xl w-32 md:block lg:w-40 lg:text-2xl xl:w-48 xl:text-3xl 2xl:border-4 2xl:w-72 2xl:text-6xl 2xl:rounded-full" key={index}>{link.name.toUpperCase()}</button>
            })
        }

      </nav>
    )
  }
  
export default Navigation