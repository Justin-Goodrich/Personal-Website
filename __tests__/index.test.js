import React from "react";
import { render, screen } from "@testing-library/react"
import renderer from "react-test-renderer"
import Home from '../pages/index'
import Navigation from '../components/Navigation'

describe("HomePage", () => {
    it('renders correctly', () => {
        const tree = renderer
          .create(<Home/>)
          .toJSON();
        expect(tree).toMatchSnapshot();
      });
  });

describe("Navigation", () => {
    it("renders all buttons",() => {
        const tree = renderer
          .create(<Navigation/>)
          .toJSON();
        expect(tree).toMatchSnapshot();
    })
})