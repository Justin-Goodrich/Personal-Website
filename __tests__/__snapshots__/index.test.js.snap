// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders correctly 1`] = `
<div>
  <div
    className="bg-offwhite"
  >
    <div
      className="hidden md:flex justify-center"
    >
      <nav
        className="hidden md:flex w-fit p-0.5 border-black border-2 rounded-3xl space-x-3 items-center 2xl:border-4 2xl:p-2 2xl:rounded-full"
      >
        <button
          className="hidden border-black border-2 rounded-3xl text-2xl w-32 md:block lg:w-40 lg:text-2xl xl:w-48 xl:text-3xl 2xl:border-4 2xl:w-72 2xl:text-6xl 2xl:rounded-full"
        >
          HOME
        </button>
        <button
          className="hidden border-black border-2 rounded-3xl text-2xl w-32 md:block lg:w-40 lg:text-2xl xl:w-48 xl:text-3xl 2xl:border-4 2xl:w-72 2xl:text-6xl 2xl:rounded-full"
        >
          SKILLS
        </button>
        <button
          className="hidden border-black border-2 rounded-3xl text-2xl w-32 md:block lg:w-40 lg:text-2xl xl:w-48 xl:text-3xl 2xl:border-4 2xl:w-72 2xl:text-6xl 2xl:rounded-full"
        >
          PROJECTS
        </button>
        <button
          className="hidden border-black border-2 rounded-3xl text-2xl w-32 md:block lg:w-40 lg:text-2xl xl:w-48 xl:text-3xl 2xl:border-4 2xl:w-72 2xl:text-6xl 2xl:rounded-full"
        >
          CONTACT
        </button>
      </nav>
    </div>
  </div>
</div>
`;

exports[`Navigation renders all buttons 1`] = `
<nav
  className="hidden md:flex w-fit p-0.5 border-black border-2 rounded-3xl space-x-3 items-center 2xl:border-4 2xl:p-2 2xl:rounded-full"
>
  <button
    className="hidden border-black border-2 rounded-3xl text-2xl w-32 md:block lg:w-40 lg:text-2xl xl:w-48 xl:text-3xl 2xl:border-4 2xl:w-72 2xl:text-6xl 2xl:rounded-full"
  >
    HOME
  </button>
  <button
    className="hidden border-black border-2 rounded-3xl text-2xl w-32 md:block lg:w-40 lg:text-2xl xl:w-48 xl:text-3xl 2xl:border-4 2xl:w-72 2xl:text-6xl 2xl:rounded-full"
  >
    SKILLS
  </button>
  <button
    className="hidden border-black border-2 rounded-3xl text-2xl w-32 md:block lg:w-40 lg:text-2xl xl:w-48 xl:text-3xl 2xl:border-4 2xl:w-72 2xl:text-6xl 2xl:rounded-full"
  >
    PROJECTS
  </button>
  <button
    className="hidden border-black border-2 rounded-3xl text-2xl w-32 md:block lg:w-40 lg:text-2xl xl:w-48 xl:text-3xl 2xl:border-4 2xl:w-72 2xl:text-6xl 2xl:rounded-full"
  >
    CONTACT
  </button>
</nav>
`;
